### PR TITLE
Allow skipping sanitization of / char in export_remote tool

### PR DESCRIPTION
**Pull request by mvdbeek**
and allow using collection name as outer directory.

I think that should help with some of the VGP export needs, but I am a little lost in the requirements for this project.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/tools/parameters/wrapped_json.py [hunk 1]
```diff
@@ -44,11 +44,12 @@ def data_collection_input_to_staging_path_and_source_path(v):
          } for staging_path, source_path, metadata_files in zip(staging_paths, source_paths, metadata_files)]
 
 
-def data_input_to_staging_path_and_source_path(v):
+def data_input_to_staging_path_and_source_path(v, invalid_chars=('/',)):
+    staging_path = v.staging_path(invalid_chars=invalid_chars)
```
Comment (nuwang) [inline lib/galaxy/tools/parameters/wrapped_json.py]:
```suggestion
    staging_path = v.get_staging_path(invalid_chars=invalid_chars)
```

File: lib/galaxy/tools/wrappers.py [hunk 2]
```diff
@@ -315,23 +315,22 @@ def file_ext(self):
     def name_and_ext(self):
         return f"{self.element_identifier}.{self.file_ext}"
 
-    @property
-    def staging_path(self):
+    def staging_path(self, invalid_chars=('/',)):
```
Comment (nuwang) [inline lib/galaxy/tools/wrappers.py]:
```suggestion
    def get_staging_path(self, invalid_chars=('/',)):
```

**Review by nuwang — APPROVED** [review]
Have tested this and can confirm that this works well (have also tested some edge cases and found no obvious issues). I agree with @mvdbeek that the ideal solution here would be to change the workflows to use nested collections if possible. Is that at all a possibility @Delphine-L? I'm guessing the answer is no, and it would be great to find out the reasons for this - is this a UI issue for example?

**Review by jmchilton — APPROVED** [review]

File: tools/data_export/export_remote.xml [hunk 3]
```diff
@@ -35,7 +35,7 @@ $fileconfig
         <conditional name="export_type">
             <param name="export_type_selector" type="select" label="What would you like to export?">
                 <option value="datasets_auto" selected="True">Datasets using their Galaxy name and extension.</option>
```
Comment (nsoranzo) [inline tools/data_export/export_remote.xml]:
```suggestion
                <option value="datasets_auto" selected="true">Datasets using their Galaxy name and extension</option>
```

File: tools/data_export/export_remote.xml [hunk 4]
```diff
@@ -35,7 +35,7 @@ $fileconfig
...
-                <option value="collection_auto">Dataset Collections using their name, structure and extension</option>
+                <option value="collection_auto">Dataset Collection using their name, structure and extension</option>
                 <option value="datasets_named">Datasets using specified name.</option>
```
Comment (nsoranzo) [inline tools/data_export/export_remote.xml]:
```suggestion
                <option value="datasets_named">Datasets specifying their names</option>
```

**Review by dannon — APPROVED** [review]
